include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -664,19 +645,19 @@ __pattern_equal(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Iterator1
     if (__last1 == __first1 || __last2 == __first2 || __last1 - __first1 != __last2 - __first2)
         return false;
 
-    using _Predicate = oneapi::dpl::unseq_backend::single_match_pred<equal_predicate<_Pred>>;
+    using _Predicate = oneapi::dpl::unseq_backend::single_match_pred<oneapi::dpl::__internal::__not_pred<_Pred>>;
 
-    auto __keep1 = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator1>();
-    auto __buf1 = __keep1(__first1, __last1);
-    auto __keep2 = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator2>();
-    auto __buf2 = __keep2(__first2, __last2);
+    auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read>();
+    auto __buf1 = __keep(__first1, __last1);
+    auto __buf2 = __keep(__first2, __last2);
+
+    using size_calc = oneapi::dpl::__par_backend_hetero::__min_size_calc;
```

Comment:
Previously, only `__parallel_find_or_min_size` used `__min_size_calc`, which in turn was used only by `__pattern_mismatch`. So, everything except for mismatch algorithm relied on `__first_size_calc`.

Is that change intentional?

Reply:
Yes, it is intentional change.
Everywhere, where we have 1-dimensional iteration space and simultaneous access to more then one range, should be "minimum size logic" used, otherwise we have (might have) got UB.